pkg/controllers/managementlegacy/globaldns/globaldnsprovider_catalog_launcher.go

```diff
@@ -276,17 +266,37 @@ func (n *ProviderCatalogLauncher) createUpdateExternalDNSApp(obj *v3.GlobalDnsPr
 	return nil, nil
 }
 
-func (n *ProviderCatalogLauncher) getProviderIfAlreadyRunning(obj *v3.GlobalDnsProvider) (*pv3.App, error) {
+func (n *ProviderCatalogLauncher) deleteExternalDNSApp(key string, obj *v3.GlobalDnsProvider) (runtime.Object, error) {
+	var err error
```

Comment:
Maybe I'm missing something obvious here, but why do we have to declare `err` here?

Reply:
cleaned up